src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -337,42 +139,39 @@ std::variant<Variable, ParsedQuery::Alias> SparqlQleverVisitor::visitTypesafe(
 }
 
 // ____________________________________________________________________________________
-ParsedQuery::Alias SparqlQleverVisitor::visitTypesafe(
-    SparqlAutomaticParser::AliasContext* ctx) {
+ParsedQuery::Alias Visitor::visitTypesafe(Parser::AliasContext* ctx) {
   // A SPARQL alias has only one child, namely the contents within
   // parentheses.
   return visitTypesafe(ctx->aliasWithoutBrackets());
 }
 
 // ____________________________________________________________________________________
-ParsedQuery::Alias SparqlQleverVisitor::visitTypesafe(
-    SparqlAutomaticParser::AliasWithoutBracketsContext* ctx) {
-  auto wrapper = makeExpressionPimpl(visit(ctx->expression()));
+ParsedQuery::Alias Visitor::visitTypesafe(
+    Parser::AliasWithoutBracketsContext* ctx) {
+  auto wrapper = makeExpressionPimpl(visitTypesafe(ctx->expression()));
   return ParsedQuery::Alias{std::move(wrapper), ctx->var()->getText()};
 }
 
 // ____________________________________________________________________________________
-Variable SparqlQleverVisitor::visitTypesafe(
-    SparqlAutomaticParser::VarContext* ctx) {
+Variable Visitor::visitTypesafe(Parser::VarContext* ctx) {
   return Variable{ctx->getText()};
 }
 
 // ____________________________________________________________________________________
-GraphPatternOperation::Bind SparqlQleverVisitor::visitTypesafe(
-    SparqlAutomaticParser::BindContext* ctx) {
-  auto wrapper = makeExpressionPimpl(visit(ctx->expression()));
+GraphPatternOperation::Bind Visitor::visitTypesafe(Parser::BindContext* ctx) {
+  auto wrapper = makeExpressionPimpl(visitTypesafe(ctx->expression()));
```

Comment:
I think if we make the  `SparqlExpressionPimpl`-constructor not explicit (it takes a &&anyway and the type is close enough, then the `makeExpressionPimpl`-function is really unneeded, and we can just write
```
return {visitTypesafe(ctx->expression()), ctx->var()->getText()};